pants

```diff
@@ -73,23 +73,54 @@ EOF
 
 # The high-level flow:
 # 1.) Resolve the Python interpreter, first reading from the env var $PYTHON,
-#     then reading from pants.ini, then defaulting to `python2.7`.
+#     then reading from pants.ini, then determining the default based off of
+#     which Python versions the current `pants_version` supports.
 # 2.) Grab pants version from pants.ini or default to latest.
 # 3.) Check for a venv via a naming/path convention and execute if found.
 # 4.) Otherwise create venv and re-exec self.
 #
 # After that pants itself will handle making sure any requested plugins
 # are installed and up to date.
 
-function determine_pants_runtime_python_version {
+function determine_default_python_exe {
+  pants_version="$(get_pants_ini_config_value 'pants_version')"
+  # If `pants_version` not pinned in `pants.ini`, default to what the most
+  # recent stable release supports.
+  if [[ -z "${pants_version}" ]]; then
+    supported_python_versions=('3.6' '2.7')
+    supported_message='2.7 or 3.6'
+  else
+    pants_minor_version="$(echo ${pants_version} | cut -d '.' -f2)"
```

Comment:
If the major version is greater than `1`, we should default to the max python versions. Just in case we ever cut `2.0.0` =)

Reply:
Perhaps leaving a TODO about `2.0.0` would be appropriate?

Reply:
I think that we know that `2.0.0` should definitely not use python `2.7`, which is the current effect of this code. I think it's a blocker, unfortunately. But it should only be two extra lines.